x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/Monitoring.java

```diff
@@ -103,10 +106,12 @@ public Monitoring(Settings settings) {
         final ClusterSettings clusterSettings = clusterService.getClusterSettings();
         final CleanerService cleanerService = new CleanerService(settings, clusterSettings, threadPool, getLicenseState());
         final SSLService dynamicSSLService = getSslService().createDynamicSSLService();
+        final MonitoringMigrationCoordinator migrationCoordinator = new MonitoringMigrationCoordinator();
```

Comment:
This instance is passed to the exporters, but I don't think this is the instance that passed into the Transport action (that one is created by Guice and injected). Which would mean that any state set by the transport actions would not be read by the exporters, which defeats the purpose of having an external coordinator.  

I wonder if we should just remove the MonitoringMigrationCoordinator class entirely in favor of boolean in the tranport action that dis-allows running that action overlapping. 

If I understand the global coordinator is there to ensure that our cluster state updates from these operations don't end up re-constructing the exporters we are trying to disable while we are disabling them ? If so, can the transport action write a small piece of cluster state and the exporters onClusterState change read that and skip reconstruction while that state exists in the cluster ?

Reply:
OK ... I missed that that Node.java does `  pluginComponents.stream().forEach(p -> b.bind((Class) p.getClass()).toInstance(p));` where plugin components are those items created by the plugins, thereby making them the same instance objects that get injected. Sorry for the noise.